tests/unit/lms/services/application_instance_service_test.py

```diff
@@ -2,65 +2,43 @@
 
-from lms.models import ApplicationInstance
+from lms.services import ConsumerKeyError
 from lms.services.application_instance import factory
 from tests import factories
 
 
-class TestApplicationInstanceGetter:
+class TestApplicationInstanceService:
     def test_get_one(self, svc, test_application_instance):
         assert (
             svc.get(test_application_instance.consumer_key) == test_application_instance
         )
 
-    def test_get_none(self, svc):
-        assert svc.get("NOPE") is None
-
-    def test_update_settings_no_update(
-        self, svc, test_application_instance, db_session
-    ):
-        svc.update_settings(test_application_instance)
-
-        # Let's re-query to get the state form the DB
-        ai = db_session.query(ApplicationInstance).get(test_application_instance.id)
-
-        assert ai.settings == test_application_instance.settings
-
-    @pytest.mark.parametrize(
-        "setting_name,setting_location",
-        [
-            ("canvas_sections_enabled", "canvas.sections_enabled"),
-            ("canvas_groups_enabled", "canvas.groups_enabled"),
-        ],
-    )
-    def test_update_settings_boolean_toggle(
-        self,
-        setting_name,
-        setting_location,
-        svc,
-        test_application_instance,
-        db_session,
-    ):
-        svc.update_settings(test_application_instance, **{setting_name: True})
-
-        # Let's re-query to get the state form the DB
-        ai = db_session.query(ApplicationInstance).get(test_application_instance.id)
-        assert ai.settings.get(*setting_location.split(".")) is True
-
-        svc.update_settings(test_application_instance, **{setting_name: False})
-
-        # Re-query again
-        ai = db_session.query(ApplicationInstance).get(test_application_instance.id)
-        assert (
-            # pylint: disable=compare-to-zero
-            ai.settings.get(*setting_location.split("."))
-            is False
-        )
+    def test_get_with_default_consumer_key(self, svc, default_application_instance):
+        assert svc.get() == default_application_instance
+
+    def test_get_raises_ConsumerKeyError_if_consumer_key_doesnt_exist(self, svc):
+        with pytest.raises(ConsumerKeyError):
+            assert svc.get("NOPE") is None
+
+    def test_get_raises_ConsumerKeyError_if_consumer_key_is_None(self, pyramid_request):
+        pyramid_request.lti_user = None
+        svc = factory(mock.sentinel.context, pyramid_request)
+
+        with pytest.raises(ConsumerKeyError):
+            svc.get(None)
 
     @pytest.fixture
     def svc(self, pyramid_request):
         return factory(mock.sentinel.context, pyramid_request)
 
+    @pytest.fixture(autouse=True)
+    def default_application_instance(self, db_session, pyramid_request):
+        ai = factories.ApplicationInstance(
+            consumer_key=pyramid_request.lti_user.oauth_consumer_key
+        )
+        db_session.flush()
+        return ai
+
     @pytest.fixture(autouse=True)
```

Comment:
Not added by you, but I think the naming here could be tidied up a bit. We have `test_application_instance`. I don't think `test` is doing any work other than to distinguish it from the `application_instances` below. We have a comment to explain that these are "noise". Using the rule of naming that says if you have to immediately explain yourself it's a bad name, I'd suggest:

* `test_application_instance` -> `application_instance` - I think this is our normal policy for things to just name it after the thing when possible
* `application_instances` -> `noise_application_instances`

Reply:
Done: https://github.com/hypothesis/lms/pull/2651/commits/63b16573ec4251e6626dbdb930252d79a591986c